src/main/scala/io/shiftleft/js2cpg/preprocessing/TranspilingEnvironment.scala

```diff
@@ -61,8 +75,24 @@ trait NpmEnvironment {
     case Some(value) =>
       value
     case None =>
-      isValid = Some(isNpmAvailable && setNpmPython())
+      isValid = Some((yarnAvailable() || npmAvailable()) && setNpmPython())
       isValid.get
   }
 
+  protected def yarnAvailable(): Boolean = isYarnAvailable match {
```

Comment:
We can let the Scala compiler generate all this lazy initialization code by declaring this as a`lazy val`. (Or is there a reason not to do this here that I'm missing?)

Reply:
Sadly, this would not work. Two options here:
1) put the `lazy val`s inside of class `TranspilingEnvironment`: as multiple instances of this require the environment to be checked we would end up with multiple invocations if yarn/npm.
2) put the `lazy val`s inside of object `TranspilingEnvironment`: the initialization code needs the `projectPath` that comes from `Transpiler` instances we would have no access to in the object.

Reply:
Ah, I see. Looks like your current solution is the best one.